huaweicloud/resource_huaweicloud_evs_volume.go

```diff
@@ -214,50 +215,35 @@ func resourceEvsVolumeV3Create(d *schema.ResourceData, meta interface{}) error {
 	return resourceEvsVolumeV3Read(d, meta)
 }
 
-func resourceEvsVolumeV3Read(d *schema.ResourceData, meta interface{}) error {
-	config := meta.(*config.Config)
-	blockStorageClient, err := config.BlockStorageV3Client(GetRegion(d, config))
-	if err != nil {
-		return fmtp.Errorf("Error creating HuaweiCloud EVS storage client: %s", err)
+func setEvsVolumeDeviceType(d *schema.ResourceData, resp *volumes.Volume) error {
+	if value, ok := resp.Metadata["hw:passthrough"]; ok && value == "true" {
+		return d.Set("device_type", "SCSI")
 	}
+	return d.Set("device_type", "VBD")
+}
 
-	v, err := volumes.Get(blockStorageClient, d.Id()).Extract()
-	if err != nil {
-		return CheckDeleted(d, err, "volume")
+func setEvsVolumeImageId(d *schema.ResourceData, resp *volumes.Volume) error {
+	if value, ok := resp.VolumeImageMetadata["image_id"]; ok {
+		return d.Set("image_id", value)
 	}
+	return nil
+}
 
-	logp.Printf("[DEBUG] Retrieved volume %s: %+v", d.Id(), v)
-
-	d.Set("name", v.Name)
-	d.Set("size", v.Size)
-	d.Set("description", v.Description)
-	d.Set("availability_zone", v.AvailabilityZone)
-	d.Set("snapshot_id", v.SnapshotID)
-	d.Set("volume_type", v.VolumeType)
-	d.Set("enterprise_project_id", v.EnterpriseProjectID)
-	d.Set("region", GetRegion(d, config))
-	d.Set("wwn", v.WWN)
-	if value, ok := v.Metadata["hw:passthrough"]; ok && value == "true" {
-		d.Set("device_type", "SCSI")
-	} else {
-		d.Set("device_type", "VBD")
-	}
-	if value, ok := v.VolumeImageMetadata["image_id"]; ok {
-		d.Set("image_id", value)
-	}
-	d.Set("multiattach", v.Multiattach)
-	// set tags
+func setEvsVolumeTags(d *schema.ResourceData, resp *volumes.Volume) error {
 	tags := make(map[string]string)
-	for key, val := range v.Tags {
+	for key, val := range resp.Tags {
```

Comment:
the type of resp.Tags is map[string]string, can we set `tags` directly?

Reply:
absolutely yes.